### ISSUE TITLE
:move may mess up fold levels

### ISSUE THREAD

**Issue by lifepillar** [vim/vim#1360]
To reproduce, use the following `test_vimrc`:

```
set shiftwidth=4
set foldmethod=expr
set foldexpr=MyFold(v:lnum)
set foldlevel=20
set foldcolumn=7

fun! MyFold(lnum)
  return '>' . (1 + indent(a:lnum) / shiftwidth())
endf
```

and run `vim -Nu test_vimrc`. Edit this file (indented with spaces):

```
aaa
    bbb
        ccc
            ddd
        eee
        fff
        ggg
            hhh
            iii
        jjj
    kkk
    lll
mmm

abc
    def
    ghi
    jkl
mno
    pqr
```

Visually select, say, lines 2–10 then type `:move 1`. Note how the fold levels change, although the text in the buffer has not changed. If the lines after, and including, `abc` are deleted, the bug does not occur.

**Comment by lifepillar** [comment]
Btw, `zx` or `zX` do not help, but executing `:undo` then `:redo` after the `:move` command restores the correct fold levels.

**Comment by luchr** [comment]
Perhaps related to Issue #687.

**Comment by chrisbra** [comment]
fixed by [v8.0.0459)](https://github.com/vim/vim/releases/tag/v8.0.0459)
